### ISSUE TITLE
[2.0.4.31] Issue: cannot create a new backup job

### ISSUE THREAD

**Issue by sfahrenholz** [duplicati/duplicati#3955]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.4.31
- **Operating system**: Windows 10
- **Backend**: Folder

## Description
After updating to 2.0.4.31_canary_2019-10-19 i want to add a new backup job. In the last step of the wizard, the information, "Unable to save schedule or backup object" are showing. The Job are not created.

## Steps to reproduce

1. Add a new Backup
2. Set the schedule plan
3. finished the backup job

## Screenshots
![image](https://user-images.githubusercontent.com/6977657/67178518-5880dd00-f3d3-11e9-80e8-27d631c781a2.png)

![image](https://user-images.githubusercontent.com/6977657/67178506-4dc64800-f3d3-11e9-8535-b2d5be435186.png)

there are no logs..

**Comment by DarkSilence** [comment]
Same here. Using latest build

**Comment by ts678** [comment]
Working here on Windows 10 on fresh Duplicati install with fresh job DB. I'm not well set up to upgrade, but I don't know if that's the key difference. Can anyone give me more details about the path you took?

**Comment by ts678** [comment]
Google Translate thinks the error message from the image is German and translates to English as "The UTC time displayed after applying the offset must be between 0 and 10000". I noticed several Google search hits suggesting that the standard English error is "The UTC time represented when the offset is applied must be between year 0 and 10,000". Someone who's more familiar with C# time should look.

Is the @DarkSilence error the same message in German, or in some other language, or very different?

**Comment by DarkSilence** [comment]
@ts678 I'll provide more details. My OS is Windows 7 (all latest patches installed). English (US). Bug appears when I create a new backup and on tab #4 I check „Automatically run backups.“ checkbox. From my searchings it doesn't matter what date\time I put below the checkbox. If the checkbox is set I get this error, when checkbox is unset there's no error. I get this message in english. It says: „The UTC time represented when the offset is applied must be between year 0 and 10,000.´“. Bug only appears only on the latest build of Duplicati. I installed the previous build and couldn't reproduce the bug there.

**Comment by DarkSilence** [comment]
@ts678  Maybe this #3919 have some connections with the bug...

**Comment by sfahrenholz** [comment]
The path of source and destination is not important in this case. 
for example
destination : C:\BackupFolder
source: C:\SourceFolder

gives the message.

Maybe it is important: this is an update installation.

**Comment by warmup72** [comment]
I´ve the same Problem with 2.0.4.31 on my mac 10.15 (Catalina).
I´ve edited an existion Backup-Job. Set him to "no schedule" after set I couldn´t aktivate the schedule anymore.
Workaround. Export to json; edit the job an import it again

**Comment by ts678** [comment]
Thanks all for input. I'm also gathering input at [Release: 2.0.4.31 (canary) 2019-10-19](https://forum.duplicati.com/t/release-2-0-4-31-canary-2019-10-19/8278/3?u=ts678). For @warmup72

> Workaround. Export to json; edit the job an import it again.

What was the edit? Actually the whole sequence is a bit unclear to me. Are you saying you changed an existing scheduled backup job to "no schedule", then a Save attempt errored about that? If not, what?

On a different subject, most of you possibly are in time zones which are "GMT+#". Are any of you not?
I'm not sure it's significant, but my export of a newly scheduled job showed LastRun that's just after the start of the .NET epoch. I'm in GMT-5 TZ. If yours tried before the start of time, that might be the error.

~~~
  "Schedule": {
    "ID": 1,
    "Tags": [
      "ID=2"
    ],
    "Time": "2019-10-23T18:00:00Z",
    "Repeat": "1D",
    "LastRun": "0001-01-01T05:00:00Z",
    "Rule": "AllowedWeekDays=Monday,Tuesday,Wednesday,Thursday,Friday,Saturday",
    "AllowedDays": [
      "mon",
      "tue",
      "wed",
      "thu",
      "fri",
      "sat"
    ]
~~~

Following the @DarkSilence theory about NormalizeDateTimeToEpochSeconds, there are some around.

https://github.com/duplicati/duplicati/blob/6e0b4bbb1324b5a8d4c679caa575cda95f221842/Duplicati/Server/Database/Connection.cs#L631-L631

Let me give a heads up to @kylestanfield and @warwickmm who worked on the time handling update.

**Comment by warmup72** [comment]
Well, I'm trying to explain what I did:
Before upgrading to macos 10.15, I turned off the timed duplicate job.
After the macos update was successful, I wanted to reactivate the daily backup of the Duplicati job.
That did not work, because I always received the above error message.
Thus, I have exported the job and entered the "scheduler sequence" in the Json job manually. Then I imported this job as a new backup job and changed the database to that of the old job.
I hope I could describe it a bit clearer

**Comment by ts678** [comment]
[The UTC time represented when the offset is applied must be between year 0 and 10,000. Parameter name: offset](https://stackoverflow.com/a/13799438) describes how what I called "really weird values" at or near the minimum or maximum times can be shifted out of the allowable range by time zone shifts when the date is converted to UTC. The old code looks like it doesn't care. The minimum .NET time (quite possibly from an uninitialized variable being 0) just turns into a large negative number of seconds from the 1970 UNIX epoch (visible in the database).

Although LastRun gets passed around quite a lot, the only thing I could see it got used for was the below:

https://github.com/duplicati/duplicati/blob/6e0b4bbb1324b5a8d4c679caa575cda95f221842/Duplicati/Server/Scheduler.cs#L250-L256

I'm not sure there is any simple workaround, so I hope this gets fixed soon. An alternative workaround to hand-crafting the Schedule for re-import might be a temporary time zone change only for job creations.

**Comment by warwickmm** [comment]
I think the change in pull request #3919 is definitely the culprit.  I suspect that when creating a new backup, the call to [AddOrUpdateSchedule](https://github.com/duplicati/duplicati/blob/6e0b4bbb1324b5a8d4c679caa575cda95f221842/Duplicati/Server/Database/Connection.cs#L614) provides an [ISchedule](https://github.com/duplicati/duplicati/blob/6e0b4bbb1324b5a8d4c679caa575cda95f221842/Duplicati/Server/Duplicati.Server.Serialization/Interface/ISchedule.cs) with uninitialized `Time` and `LastRun` properties, which will default to `DateTime.MinValue`.  This can cause the issue described in the link that @ts678 posted:

> "Any DateTimeOffset value is converted to Coordinated Universal Time (UTC) before the method performs the comparison with MinValue. This means that a DateTimeOffset value whose date and time are close to the minimum range, but whose offset is positive, may throw an exception. For example, the value 1/1/0001 1:00:00 AM +02:00 is out of range because it is one hour earlier than MinValue when it is converted to UTC."

**Comment by warwickmm** [comment]
I think the simplest solution is to revert the changes from #3919.  See pull request #3960.

**Comment by warwickmm** [comment]
Fixed in pull request #3960.
